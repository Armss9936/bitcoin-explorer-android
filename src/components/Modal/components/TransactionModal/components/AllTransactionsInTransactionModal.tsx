import uuid from "react-native-uuid";
import { View, Text, StyleSheet, ScrollView } from "react-native";
import { TransactionType } from "src/components/Modal/types";
import { I18nContext } from "src/context/LocaleProvider";
import { useContext } from "react";

type AllTransactionsInTransactionModalProps = {
  inputTransactions: TransactionType["inputTransactions"];
  outputTransactions: TransactionType["outputTransactions"];
};

export default function AllTransactionsInTransactionModal(
  props: AllTransactionsInTransactionModalProps
) {
  const { inputTransactions, outputTransactions } = props;
  const i18nContext = useContext(I18nContext);

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{i18nContext.t("inputs_and_outputs")}</Text>

      <View style={styles.transactionContainer}>
        <View style={styles.inputTransactionsContainer}>
          {inputTransactions.map((inputTransaction) => {
            const { prevout } = inputTransaction;

            return (
              <View
                style={styles.transactionContentContainer}
                key={`${uuid.v4()}`}
              >
                <Text style={styles.transactionAddress}>
                  {(prevout?.scriptpubkey_address)
                    .toString()
                    .slice(0, 14)
                    .concat("...")}
                </Text>

                <Text style={styles.text}>
                  {(prevout?.value / 100000000).toString()} BTC
                </Text>
              </View>
            );
          })}
        </View>

        <Text style={styles.destinationTransactionSeparator}>{">"}</Text>

        <View style={styles.outputTransactionsContainer}>
          {outputTransactions.map((outputTransaction) => {
            const { value, scriptpubkey_address } = outputTransaction;

            return (
              <View
                key={`${uuid.v4()}`}
                style={styles.eachTransactionOutputContainer}
              >
                <Text style={styles.transactionAddress}>
                  {scriptpubkey_address.toString().slice(0, 14).concat("...")}
                </Text>
                <Text style={styles.text}>{value / 100000000} BTC</Text>
              </View>
            );
          })}
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  title: {
    color: "#FFF",
    fontSize: 15,
    fontWeight: "600",
  },
  container: {
    gap: 16,
    height: "100%",
    backgroundColor: "#101427",
  },
  transactionContainer: {
    backgroundColor: "#1d2133",
    borderRadius: 8,
    paddingVertical: 5,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-evenly",
    maxWidth: "100%",
  },
  inputTransactionsContainer: {
    gap: 5,
    alignItems: "center",
  },
  outputTransactionsContainer: {
    gap: 5,
    alignItems: "center",
  },
  eachTransactionOutputContainer: {
    width: "100%",
    alignItems: "center",
  },
  transactionContentContainer: {
    alignItems: "center",
    width: "100%",
  },
  transactionAddress: {
    color: "#DF7800",
    fontSize: 16,
  },
  text: {
    color: "#FFF",
    fontSize: 13,
    textAlign: "right",
  },
  destinationTransactionSeparator: {
    color: "#D9D9D9",
    fontSize: 28,
    fontWeight: "800",
  },
});
