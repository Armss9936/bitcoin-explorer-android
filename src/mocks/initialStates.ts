import { TransactionType } from "src/components/Modal/types";

const timeAgoInitialState = {
  hour: 0,
  minutes: 0,
  day: 0,
  month: 0,
  year: 0,
};

export const initialStateBlocks = [
  {
    blockHeight: 1,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
  {
    blockHeight: 2,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
  {
    blockHeight: 3,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
  {
    blockHeight: 4,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
];

export const initialStateTransaction: TransactionType = {
  transactionId: " ",
  size: 0,
  fee: 0,
  inputTransactions: [
    {
      prevout: {
        scriptpubkey_address: " ",
        value: 0,
      },
    },
  ],
  outputTransactions: [
    {
      scriptpubkey_address: " ",
      value: 0,
    },
  ],
  statusTransaction: undefined,
};
