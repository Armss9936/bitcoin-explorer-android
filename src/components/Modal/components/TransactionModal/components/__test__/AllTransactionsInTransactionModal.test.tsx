/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import { initialStateTransaction } from "src/__mocks__/initialStates";
import AllTransactionsInTransactionModal from "../AllTransactionsInTransactionModal";

jest.mock("src/stores/App/useAppStore", () => ({
  __esModule: true,
  default: jest.fn(() => ({
    localization: { t: (key) => key }, // Mock da função t de tradução
    selectedCurrency: "USD", // Moeda mockada
    bitcoinPrice: {
      mempool: { USD: 50000 },
      blockchainInfo: { USD: { buy: 50000 } },
    }, // Preço fictício
  })),
}));

describe("AllTransactionsInTransactionModal", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <AllTransactionsInTransactionModal
          inputTransactions={initialStateTransaction.inputTransactions}
          outputTransactions={initialStateTransaction.outputTransactions}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
