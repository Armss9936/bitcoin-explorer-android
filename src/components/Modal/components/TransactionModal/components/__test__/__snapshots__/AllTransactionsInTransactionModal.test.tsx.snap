// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllTransactionsInTransactionModal renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#101427",
      "gap": 16,
      "height": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#FFF",
        "fontSize": 15,
        "fontWeight": "600",
      }
    }
  >
    inputs_and_outputs
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#1d2133",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "maxWidth": "100%",
        "paddingVertical": 5,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "gap": 5,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 16,
            }
          }
        >
           ...
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 13,
              "textAlign": "right",
            }
          }
        >
          0
           BTC
        </Text>
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 14,
              "textAlign": "center",
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        {
          "color": "#D9D9D9",
          "fontSize": 28,
          "fontWeight": "800",
        }
      }
    >
      &gt;
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 5,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 16,
            }
          }
        >
           ...
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 13,
              "textAlign": "right",
            }
          }
        >
          0
           BTC
        </Text>
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 14,
              "textAlign": "center",
            }
          }
        >
          $ 0,00
        </Text>
      </View>
    </View>
  </View>
</View>
`;
